Source/Core/DolphinQt/EditTimePlayedDialog.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright 2025 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "DolphinQt/EditTimePlayedDialog.h"
+
+#include <chrono>
+#include <string>
+
+#include <QDialog>
+#include <QDialogButtonBox>
+#include <QFormLayout>
+#include <QLabel>
+#include <QSpinBox>
+#include <QString>
+#include <QWidget>
+
+#include "Core/TimePlayed.h"
+#include "Core/TitleDatabase.h"
+
+#include "UICommon/GameFile.h"
+
+EditTimePlayedDialog::EditTimePlayedDialog(QWidget* const parent, const UICommon::GameFile& game)
+    : QDialog(parent), m_game_id(game.GetGameID())
+{
+  using namespace std::chrono_literals;
+
+  // std::chrono::milliseconds is guaranteed to have a signed integer representation type with at
+  // least 45 bits (44 non-sign bits). (2^44)/(3'600'000 milliseconds/hour) ~= 4'886'718 hours.
+  // Setting MAX_EDITABLE_HOURS to 999'999 makes all <= 6 digit hour counts valid and corresponds to
+  // 114 years, which is more than anyone will actually ever play.
+  constexpr auto MAX_EDITABLE_HOURS = std::chrono::hours(999'999);
+  constexpr std::chrono::milliseconds MAX_EDITABLE_TIME =
+      MAX_EDITABLE_HOURS + std::chrono::minutes(59);
+
+  const std::chrono::milliseconds time_played =
+      std::min(MAX_EDITABLE_TIME, TimePlayedManager::GetInstance().GetTimePlayed(m_game_id));
+  const auto hours_played = time_played / 1h;
+  const auto minutes_played = (time_played % 1h) / 1min;
+
+  setWindowTitle(QStringLiteral("Edit Time Played"));
+
+  auto* const layout = new QFormLayout;
+
+  m_hours_edit = new QSpinBox;
+  m_hours_edit->setRange(0, MAX_EDITABLE_HOURS.count());
+  m_hours_edit->setValue(hours_played);
+
+  m_minutes_edit = new QSpinBox;
+  m_minutes_edit->setRange(0, 59);
```

Comment:
Up to you, but I suppose you could allow users to type more than 59 minute and it wouldn't really break anything.

They can then type 120 minutes for 2 hours or whatever.

Reply:
There's also `QDoubleSpinBox` if you wanted to allow users to type "5.5 hours" or partial minutes.

Reply:
That would work. I went with 59 since it's the least surprising limit, but I could make it bigger to allow your example. Do you think I should just make it 999'999 to be consistent with the hours?

Reply:
> That would work. I went with 59 since it's the least surprising limit, but I could make it bigger to allow your example. Do you think I should just make it 999'999 to be consistent with the hours?

Sounds fine.

Reply:
Users can now set up to 999'999 minutes in addition to 999'999 hours. I decided to skip fractional minutes since (without the debug review commit) users can't see less than a minute anyway.